packages/node-sdk/src/client.ts

```diff
@@ -0,0 +1,607 @@
+    );
+    ok(
+      options.host === undefined ||
+        (typeof options.host === "string" && options.host.length > 0),
+      "host must be a string",
+    );
+    ok(
+      options.logger === undefined || isObject(options.logger),
+      "logger must be an object",
+    );
+    ok(
+      options.httpClient === undefined || isObject(options.httpClient),
+      "httpClient must be an object",
+    );
+    ok(
+      options.fallbackFlags === undefined || isObject(options.fallbackFlags),
+      "fallbackFlags must be an object",
+    );
+
+    const flags =
+      options.fallbackFlags &&
+      Object.entries(options.fallbackFlags).reduce(
+        (acc, [key, value]) => {
+          acc[key] = {
+            key,
+            value: value as boolean,
+          };
+          return acc;
+        },
+        {} as Record<string, Flag>,
+      );
+
+    this._shared = {
+      logger:
+        options.logger && decorateLogger(BUCKET_LOG_PREFIX, options.logger),
+      host: options.host || API_HOST,
+      headers: {
+        "Content-Type": "application/json",
+        [SDK_VERSION_HEADER_NAME]: SDK_VERSION,
+        ["Authorization"]: `Bearer ${options.secretKey}`,
+      },
+      httpClient: options.httpClient || fetchClient,
+      refetchInterval: FLAGS_REFETCH_MS,
+      staleWarningInterval: FLAGS_REFETCH_MS * 5,
+      fallbackFlags: flags,
+    };
+  }
+
+  /**
+   * Sends a POST request to the specified path.
+   *
+   * @param path - The path to send the request to.
+   * @param body - The body of the request.
+   * @returns A boolean indicating if the request was successful.
+   * @throws An error if the path or body is invalid.
+   **/
+  private async post<TBody>(path: string, body: TBody) {
+    ok(typeof path === "string" && path.length > 0, "path must be a string");
+    ok(typeof body === "object", "body must be an object");
+
+    try {
+      const response = await this._shared.httpClient.post<
+        TBody,
+        { success: boolean }
+      >(`${this._shared.host}/${path}`, this._shared.headers, body);
+
+      this._shared.logger?.debug(`post request to "${path}"`, response);
+      return response.body?.success === true;
+    } catch (error) {
+      this._shared.logger?.error(
+        `post request to "${path}" failed with error`,
+        error,
+      );
+      return false;
+    }
+  }
+
+  /**
+   * Sends a GET request to the specified path.
+   *
+   * @param path - The path to send the request to.
+   * @returns The response from the server.
+   * @throws An error if the path is invalid.
+   **/
+  private async get<TResponse>(path: string) {
+    ok(typeof path === "string" && path.length > 0, "path must be a string");
+
+    try {
+      const response = await this._shared.httpClient.get<
+        TResponse & { success: boolean }
+      >(`${this._shared.host}/${path}`, this._shared.headers);
+
+      this._shared.logger?.debug(`get request to "${path}"`, response);
```

Comment:
I got this when using an invalid secret key:
```
yarn start
Server is running on port 3000
[Bucket] get request to "flags" { status: 403, body: undefined }
[Bucket] received undefined value from function
Bucket initialized
```

I think the `body` is getting lost? We normally give a good error message in the ingest.

Reply:
Yes, I do not parse the body in such cases, figured a good old `403` is enough. But I can add the message as well.